cmd/samples/cron/main.go

```diff
@@ -1,66 +1,125 @@
 // The workers are supposed to be long running.
-func startWorkers(h *common.SampleHelper) {
-	// Configure worker options.
+func startWorker() {
 	workerOptions := worker.Options{
-		MetricsScope: h.Scope,
-		Logger:       h.Logger,
+		Logger: logger,
+	}
+
+	worker, err := worker.New(DomainName, TaskListName, workerOptions)
+	if err != nil {
+		logger.Fatal("Unable to create worker", zap.Error(err))
+	}
+
+	worker.RegisterWorkflow(SampleCronWorkflow)
+	worker.RegisterActivity(sampleCronActivity)
+
+	err = worker.Start()
+	if err != nil {
+		logger.Error("Unable to start worker", zap.Error(err))
 	}
-	h.StartWorkers(h.Config.DomainName, ApplicationName, workerOptions)
 }
 
-//
-// To start instance of the workflow.
-//
-func startWorkflow(h *common.SampleHelper, cron string) {
+// Start instance of the workflow.
+func startWorkflow(cron string) {
 	// This workflow ID can be user business logic identifier as well.
 	workflowID := "cron_" + uuid.New()
 	workflowOptions := client.StartWorkflowOptions{
 		ID:                              workflowID,
-		TaskList:                        ApplicationName,
+		TaskList:                        TaskListName,
 		ExecutionStartToCloseTimeout:    time.Minute,
 		DecisionTaskStartToCloseTimeout: time.Minute,
 		CronSchedule:                    cron,
 	}
-	h.StartWorkflow(workflowOptions, SampleCronWorkflow)
+
+	client, err := client.NewClient(DomainName, client.Options{})
+	if err != nil {
+		logger.Fatal("Unable to build client", zap.Error(err))
+		panic(err)
+	}
+
+	we, err := client.ExecuteWorkflow(context.Background(), workflowOptions, SampleCronWorkflow)
+	if err != nil {
+		logger.Error("Unable to execute workflow", zap.Error(err))
+	} else {
+		logger.Info("Started workflow", zap.String("WorkflowID", we.GetID()), zap.String("RunID", we.GetRunID()))
+	}
 }
 
 func main() {
+	createLogger()
+	createDomain()
+
 	var mode string
 	var cron string
 	flag.StringVar(&mode, "m", "trigger", "Mode is worker or trigger.")
 	flag.StringVar(&cron, "cron", "* * * * *", "Crontab schedule. Default \"* * * * *\"")
 	flag.Parse()
 
-	var h common.SampleHelper
-	h.SetupServiceConfig()
-
 	switch mode {
 	case "worker":
-		startWorkers(&h)
-
+		startWorker()
 		// The workers are supposed to be long running process that should not exit.
-		// Use select{} to block indefinitely for samples, you can quit by CMD+C.
+		// Use select{} to block indefinitely for samples, you can quit by Ctrl+C.
 		select {}
 	case "trigger":
-		startWorkflow(&h, cron)
+		startWorkflow(cron)
+	}
+}
+
+func createLogger() {
+	var err error
+	logger, err = zap.NewDevelopment()
+	if err != nil {
+		panic(err)
+	}
+}
+
+func createDomain() {
+	domainClient, err := client.NewDomainClient(client.Options{})
+
+	if err != nil {
+		logger.Fatal("Unable to create domain client", zap.Error(err))
+	}
+
+	err = domainClient.Register(context.Background(), &workflowservice.RegisterDomainRequest{
```

Comment:
For some reason it was just `DescribeDomain`. I don't see any issues creating domain right from here.